File: problems/1399-count-largest-group/solution.java
class Solution {
    public int countLargestGroup(int n) {
        int[] count = new int[37];
        int maxCount = 0;
        for (int i = 1; i <= n; i++) {
            int sum = 0;
            int num = i;
            while (num > 0) {
                sum += num % 10;
                num /= 10;
            }
            count[sum]++;
            maxCount = Math.max(maxCount, count[sum]);
        }
        int result = 0;
        for (int i = 1; i < 37; i++) {
            if (count[i] == maxCount) {
                result++;
            }
        }
        return result;
    }
}

File: problems/1399-count-largest-group/solution.py
class Solution:
    def countLargestGroup(self, n: int) -> int:
        groups = {}
        for i in range(1, n + 1):
            s = 0
            temp = i
            while temp > 0:
                s += temp % 10
                temp //= 10
            if s not in groups:
                groups[s] = 0
            groups[s] += 1

        max_size = 0
        count = 0
        for size in groups.values():
            if size > max_size:
                max_size = size
                count = 1
            elif size == max_size:
                count += 1

        return count